contracts/btc-finality/src/finality.rs

```diff
@@ -787,37 +785,45 @@ pub fn compute_active_finality_providers(
     })?;
 
     // Save the new set of active finality providers
-    // TODO: Purge old (height - finality depth) FP_SET entries to avoid bloating the storage (#124)
-    FP_SET.save(deps.storage, env.block.height, &finality_providers)?;
+    for (fp_btc_pk_hex, power) in fp_power_table {
```

Comment:
It's more efficient to have all active FPs in a set; basically because they are being loaded all at the same time, and saved all at the same time.
The only reason for using a map is, if you want to query the power of a given (active) FP (at a given height).